src/services/TxHistory/helpers.tsx

```diff
@@ -54,10 +79,16 @@ export const makeTxReceipt = (
 
 export const merge = (apiTxs: ITxReceipt[], accountTxs: ITxReceipt[]): ITxReceipt[] => {
   // Prioritize Account TX - needs to be more advanced?
-  const filteredApiTxs = apiTxs.filter(
-    (tx) => !accountTxs.find((a) => isSameHash(a.hash, tx.hash))
-  );
-  return filteredApiTxs.concat(accountTxs);
+
+  const apiTxsHashMap = apiTxs.reduce((acc, cur) => {
+    acc[cur.hash.toLowerCase()] = cur
```

Comment:
Will prefer for all stuff like this:
`return { ...acc, [cur.hash.toLowerCase()]: cur }`